src/domainobjects/swap_position.py

```diff
@@ -2,53 +2,123 @@
 from datetime import datetime, timedelta
 import random
 import string
+import timeit
 import pandas as pd
+import logging
 
 class SwapPosition(Generatable):
-    
-    def generate(self, record_count, custom_args):        
-        swap_contracts = self.cache.retrieve_from_cache('swap_contracts')
+
+    def generate(self, record_count, custom_args):
```

Comment:
This method requires some comments to explain the logic being applied. It is fairly hard to follow